modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase

```diff
@@ -14,6 +14,635 @@ definition {
+				targetFragmentName = "Add to Cart");
+
+			CommerceDisplayPageTemplates.addFragment(
+				collectionName = "Product",
+				fragmentName = "Availability",
+				index = 2,
+				targetFragmentName = "Price");
+		}
+
+		task ("And an account with a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("2 UOMs are created to MIN55861 with 0.6 as incremental order quantity") {
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 20,
+				incrementalOrderQuantity = "0.6",
+				productSKU = "MIN55861",
+				uomKey = "UOM1KEY",
+				uomName = "UOM1");
+
+			CommerceJSONSKUsAPI._addCommerceUOM(
+				basePrice = 20,
+				incrementalOrderQuantity = "0.6",
+				productSKU = "MIN55861",
+				promoPrice = 15,
+				uomKey = "UOM2KEY",
+				uomName = "UOM2");
+		}
+
+		task ("And from the MIN55861 product configuration, multiple quantity is set as 0.6") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				allowBackOrder = "true",
+				multipleOrderQuantity = "0.6",
+				productName = "U-Joint");
+		}
+
+		task ("When the buyer user logs in ") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("Then assert U-Joint product card shows the"View All Variants" button instead of the add to cart") {
+			CommerceNavigator.searchEntry(entryName = "U-Joint");
+
+			AssertElementPresent(
+				key_text = "View all variants",
+				locator1 = "Button#ANY");
+		}
+
+		task ("When the buyer navigates to Minium catalog → MIN55861 product details") {
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "U-Joint");
+		}
+
+		task ("And assert that the information shown on the page is related to the selected SKU/UOM(availability, stock quantity, list price, promo price)") {
+			FrontStore.assertProductDetails(
+				productAvailability = "Available",
+				productListPrice = "20.00");
+		}
+
+		task ("And adds to the cart the SKU/first UOM combination with quantity 3") {
+			Type(
+				locator1 = "CommerceAccelerators#GENERAL_QUANTITY_SELECTOR_INPUT_FIELD_ENABLED",
+				value1 = 3);
+
+			Click(locator1 = "CommerceAccelerators#GENERAL_ADD_TO_CART_BUTTON_ENABLED");
+		}
+
+		task ("And selects the second UOM") {
+			Select(
+				locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECT",
+				value1 = "UOM2");
+		}
+
+		task ("And assert that the information shown on the page is related to the selected SKU/UOM(availability, stock quantity, list price, promo price)") {
+			FrontStore.assertProductDetails(
```

Comment:
Add checks for stock quantity(add a Dynamic Field showing Inventory) and Promo Price

Thanks!